src/main/scala/org/constellation/primitives/Genesis.scala

```diff
@@ -1,112 +1,77 @@
-    implicit dao: DAO
-  ): CheckpointBlock =
-    CheckpointBlock.createCheckpointBlock(createGenesisTransaction(keyPair, allocAccountBalances), GenesisTips)(keyPair)
-
-  def start()(implicit dao: DAO): Unit = {
-    // TODO: Remove initial distribution
-    val fakeIdToGenerateTips = KeyUtils.makeKeyPair().getPublic.toId
-    val go = createGenesisAndInitialDistributionDirect(
-      dao.selfAddressStr,
-      Set(fakeIdToGenerateTips),
-      dao.keyPair,
-      dao.nodeConfig.allocAccountBalances
-    )
-    acceptGenesis(go, setAsTips = true)
+  private def createDistributionTransactions[F[_]: Concurrent](
+    allocAccountBalances: Seq[AccountBalance]
+  )(implicit dao: DAO): F[List[Transaction]] = LiftIO[F].liftIO {
+    allocAccountBalances.toList
+      .traverse(ab => dao.transactionService.createTransaction(Coinbase, ab.accountHash, ab.balance, CoinbaseKey))
   }
 
-  // TODO: Get rid of this, need to add edge case for handling tips at beginning to avoid this
-  def createDistribution(
-    selfAddressStr: String,
-    ids: Seq[Id],
-    genesisSOE: SignedObservationEdge,
-    keyPair: KeyPair
-  )(implicit dao: DAO): CheckpointBlock = {
-
-    val distr = ids.toList.traverse { id =>
-      dao.transactionService.createTransaction(selfAddressStr, id.address, 1e6.toLong, keyPair)
-    }.unsafeRunSync
-
-    CheckpointBlock.createCheckpointBlock(
-      distr,
-      Seq(
-        TypedEdgeHash(genesisSOE.hash, EdgeHashType.CheckpointHash, Some(genesisSOE.baseHash)),
-        TypedEdgeHash(genesisSOE.hash, EdgeHashType.CheckpointHash, Some(genesisSOE.baseHash))
-      )
-    )(keyPair)
-
+  private def createGenesisBlock(transactions: Seq[Transaction]): CheckpointBlock =
+    CheckpointBlock.createCheckpointBlock(transactions, GenesisTips)(CoinbaseKey)
+
+  private def createEmptyBlockFromGenesis[F[_]: Concurrent](
+    genesisSOE: SignedObservationEdge
+  )(implicit dao: DAO): F[CheckpointBlock] = LiftIO[F].liftIO {
+    val dummyTransactionSrc = KeyUtils.makeKeyPair.getPublic.toId.address
+    val dummyTransactionDst = KeyUtils.makeKeyPair.getPublic.toId.address
+
+    dao.transactionService.createDummyTransaction(dummyTransactionSrc, dummyTransactionDst, CoinbaseKey).map { tx =>
+      CheckpointBlock.createCheckpointBlock(
+        Seq(tx),
+        Seq(
+          TypedEdgeHash(genesisSOE.hash, EdgeHashType.CheckpointHash, Some(genesisSOE.baseHash)),
+          TypedEdgeHash(genesisSOE.hash, EdgeHashType.CheckpointHash, Some(genesisSOE.baseHash))
+        )
+      )(CoinbaseKey)
+    }
   }
 
-  /**
-    * Build genesis tips and example distribution among initial nodes
-    *
-    * @param ids: Initial node public keys
-    * @return : Resolved edges for state update
-    */
-  def createGenesisAndInitialDistributionDirect(
-    selfAddressStr: String,
-    ids: Set[Id],
-    keyPair: KeyPair,
+  // TODO: Make F[Unit]
+  def createGenesisObservation(
```

Comment:
@marcinwadon could you take a look at it? Adding empty transactions to empty blocks was async so I needed to make that async too. For now I left there IO to get rid of adding `unsafeRunSync` in many unit tests